utils/upodman/Makefile

```diff
@@ -0,0 +1,106 @@
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=upodman
+PKG_VERSION:=0.98.1
+PKG_RELEASE:=1
+
+PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
+PKG_SOURCE_URL:=https://codeload.github.com/oskarirauta/upodman/tar.gz/v$(PKG_VERSION)?
+PKG_HASH:=c05eddfe584fcb4e7e1a8893d676356b6332fd3ee6d292793097d16dcc33e6b9
+
+PKG_MAINTAINER:=Oskari Rauta <oskari.rauta@gmail.com>
+PKG_LICENSE:=MIT
+PKG_LICENSE_FILES:=LICENSE
+
+include $(INCLUDE_DIR)/package.mk
+
+define Package/upodman
+  SECTION:=utils
+  CATEGORY:=Utilities
+  TITLE:=Podman manager via Ubus
+  URL:=https://github.com/oskarirauta/upodman
+  DEPENDS:=+libjson-c +libstdcpp +libubox +libuci +libblobmsg-json +libubus +podman
+endef
+
+define Package/upodman/description
+  upodman extends ubus with bindings to podman and allows
+  controlling and statistics over pods and containers through
+  ubus.
+
+  upodman is written in C++ except for ubus parts which are
+  externed from C. It uses staticly linked libcurl for unix
+  socket operations, because system's default libcurl might
+  not support unix sockets.
+endef
+
+MAKE_VARS += \
+	BUILD_DIR="$(PKG_BUILD_DIR)"
+
+CURL_VERSION:=7.87.0
+CURL_PATH:=$(PKG_BUILD_DIR)/curl-$(CURL_VERSION)
+CONFIGURE_PATH:=curl-$(CURL_VERSION)
+
+CONFIGURE_ARGS+= \
+		--disable-debug --enable-optimize --disable-curldebug --disable-ares \
+		--enable-rt --disable-ech --enable-shared=no --enable-static=yes --enable-http \
+		--disable-ftp --disable-file --disable-ldap --disable-ldaps --disable-rtsp \
+		--disable-proxy --disable-dict --disable-telnet --disable-tftp --disable-pop3 \
+		--disable-imap --disable-smb --disable-smtp --disable-gopher --disable-mqtt \
+		--disable-manual --disable-ipv6 --enable-pthreads --enable-verbose --disable-sspi \
+		--disable-crypto-auth --disable-ntlm --disable-ntlm-wb --disable-tls-srp \
+		--enable-unix-sockets --disable-cookies --disable-socketpair --disable-http-auth \
+		--disable-doh --disable-mime --disable-dateparse --disable-netrc --disable-progress-meter \
+		--disable-dnsshuffle --enable-get-easy-options --disable-alt-svc --disable-headers-api \
+		--disable-hsts --disable-websockets --without-schannel --without-secure-transport \
+		--without-amissl --without-ssl --without-zlib --without-brotli --without-zstd \
+		--without-gssapi --without-ca-bundle --without-libpsl --without-libgsasl \
+		--without-libssh2 --without-libssh --without-wolfssh --without-librtmp \
+		--without-winidn --without-libidn2 --without-nghttp2 --without-ngtcp2 \
+		--without-nghttp3 --without-quiche --without-msh3 \
+		--without-zsh-functions-dir --without-fish-functions-dir
+
+define Build/Configure
+endef
+
+define Build/Prepare
+	$(call Build/Prepare/Default)
+	mkdir -p $(CURL_PATH)
+	$(TAR) xzf $(PKG_BUILD_DIR)/dl/curl-$(CURL_VERSION).tar.gz -C $(PKG_BUILD_DIR)
+endef
```

Comment:
I don't like it. This makes hard dependency on curl.

Reply:
It does. Default curl (and self-built, if opted out), does not include support for unix sockets. It's required. I first had my own socket handling, but it broke on Podman 3 -> podman 4. Curl is included in source as a "internal" curl. It's linked statically.

Reply:
Look further; that isn't openwrt dl dir, it's a directory under software's source code that just happens to be named as dl.

Reply:
I see. I still don't like it. I don't know, why there should be two versions of curl and also you are compiling it twice. Normal curl and then yours curl. So you are not only adding your's forked package, which isnt in any GNU/Linux distribution, but also we are slowing down build system by compiling embedded curl in your package. :( It also increases the build resources (capacity, etc.).

Reply:
Because provided one doesn't offer a feature needed? My curl is no way forked, it is as official as any, downloaded and stored as-is directly from curl's site. There's something wrong in what you are saying, this builds in way less than minute on decent hardware, very much identical time to so many other software. Like I previously said, read the source first, check curl's tarball and verify before assumptions.

Reply:
>  it is as official as any, downloaded and stored as-is directly from curl's site. 

How anyone can know that? You did not provide any hash in your repository. 

> read the source first,

Hm, thank you for your suggestion, but no. If you are going to argument like that, then we are not going anywhere else.

Reply:
Would it ease your mind if I'd include a curl-7.87.0.tar.gz.md5 or curl-7.87.0.tar.gz.sha256 file to the release, which would contain hash of file then?

Reply:
Personally I think we're at a point where "we" (OpenWrt) needs to decide whether to micromanage package and hunt kbytes or just try to adapt sane defaults. I'm fully aware that old MIPS devices still are "supported" but at some point we need to move on rather than holding things back with little to no benefit overall.